### ISSUE TITLE
DSL.arrayGet() does not generate required parentheses

### ISSUE THREAD

**Issue by ionutRD** [jOOQ/jOOQ#12480]
### Expected behavior

```kotlin
select(arrayGet(arrayAgg(TABLE_NAME.ID).orderBy(TABLE_NAME.UPDATED_AT.desc()), 1).from(TABLE_NAME)
```
should produce
```sql
SELECT (array_agg(id ORDER BY updated_at DESC))[1] FROM table_name;
```

### Actual behavior

```sql
SELECT array_agg(id ORDER BY updated_at DESC)[1] FROM table_name;
```

### Versions

- jOOQ: 3.14.15
- Java: 11.0.12+7
- Database (include vendor): postgres 13.3
- OS: macOS Big Sur 
- JDBC Driver (include name if inofficial driver):

**Comment by lukaseder** [comment]
Thanks a lot for your report. I think it's safe to say that almost all array value access by index could profit from such parens, e.g. even `SELECT (ARRAY[1])[1]`, which  also currently doesn't work.

**Comment by lukaseder** [comment]
Just like with `NOT(...)` (see https://github.com/jOOQ/jOOQ/issues/7362), there might occasionally be an unnecessary set of parentheses generated, such as e.g. `SELECT ((ARRAY[1] || ARRAY[2]))[1]`

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0, 3.15.4 (https://github.com/jOOQ/jOOQ/issues/12482), and 3.14.16 (https://github.com/jOOQ/jOOQ/issues/12483)

**Comment by lukaseder** [comment]
This fix has caused a regression when updating array elements, see https://github.com/jOOQ/jOOQ/issues/13808
